ocaml/xapi/xapi_cluster_helpers.ml

```diff
@@ -112,27 +112,39 @@ let corosync3_enabled ~__context =
 let maybe_generate_alert ~__context ~num_hosts ~hosts_left ~hosts_joined ~quorum
     =
   let generate_alert join cluster_host =
+    let generate_alert_body host num_hosts quorum join =
+      let num_hosts = string_of_int num_hosts in
+      let quorum = string_of_int quorum in
+      let msg =
+        if join then
+          "<message>Host has joined the cluster</message>"
+        else
+          "<message>Host has left the cluster</message>"
+      in
+      String.concat ""
+        [
+          msg
```

Comment:
Maybe inline the `msg` case here? I generally like the use of `String.concat` to create the message.

Reply:
You mean put the string literal here? The message is constructed based on the `join` parameter, I feel inlining would cause a bit of code duplication though

Reply:
The `if` expression could be inside the `String.concat`  - was my idea.